main.go

```diff
@@ -72,3 +72,45 @@ func stripPath(path string) string {
 	dir := filepath.Dir(path)
 	return strings.TrimPrefix(dir, build.Default.GOPATH+"/src/")
 }
+
+// safeRename safely moves files across different file systems.
+func safeRename(src, dest string) error {
+        if err := os.Rename(src, dest); err != nil {
+                if _, ok := err.(*os.LinkError); ok {
+                        // Looks like we are trying to move files across system volumes. Try to copy/delete strategy.
+                        tmpPath := dest + ".new"
+
+                        // Open temporary file for writing.
+                        tmpFile, err := os.OpenFile(tmpPath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0755)
```

Comment:
You are correct. I rewrote this routine using ioutil.TempFile to avoid any possible collisions with the existing files. I also added missing "io" import for io.Copy.